src/main/java/com/purbon/kafka/topology/roles/CCloudAclsProvider.java

```diff
@@ -80,27 +79,25 @@ private TopologyAclBinding translateIfNecessary(TopologyAclBinding binding) thro
             + "Service Account names into ID(s). At some point in time this will not be required anymore, "
             + "so you can configure this out by using ccloud.service_account.translation.enabled=false (true by default)");
 
-    String principal = binding.getPrincipal();
-    Long translatedPrincipalId = lookupServiceAccountId.get(principal);
-    if (translatedPrincipalId == null) { // Translation failed, so we can't continue
+    Principal principal = Principal.fromString(binding.getPrincipal());
+    long numericServiceAccountId = serviceAccountIdByNameMap.getOrDefault(principal.getServiceAccountName(), SERVICE_ACCOUNT_NOT_FOUND);
+    if (numericServiceAccountId == SERVICE_ACCOUNT_NOT_FOUND) { // Translation failed, so we can't continue
       throw new IOException(
           "Translation of principal "
               + principal
               + " failed, please review your system configuration");
     }
-    String[] fields = principal.split(":");
-
-    if (!asList("group", "user").contains(fields[0].toLowerCase())) {
-      throw new IOException("Unknown principalType: " + fields[0]);
-    }
 
     TopologyAclBinding translatedBinding =
         TopologyAclBinding.build(
             binding.getResourceType(),
             binding.getResourceName(),
             binding.getHost(),
             binding.getOperation(),
-            mappedPrincipal(fields[0], principal, translatedPrincipalId),
+            mappedPrincipal(
+                principal.getPrincipalType().name(),
```

Comment:
as a model for the principal is introduced, do you mind wrapping this within the principal class? would be much mode cleaner, right?